src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1153,6 +1153,16 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check the redirectUri before sending the request
+            //if the redirectUri from the client does not match the valid redirectUri
+            // the client app would not jump to the login page
+            if(!request.getRedirectUri().equals(this.getRedirectUriForBroker())){
+             String msg = "the redirectUri is invalid.";
```

Comment:
I expect the message to be a little bit more concrete. Can we check the following: 1) if it's starting with msauth:// 2) is the package name as expected 3) if the signature as expected? And then we can give more detailed messsage, telling which part of the redirect uri is wrong.